controller/volume_attachment_controller.go

```diff
@@ -490,7 +490,7 @@ func (vac *VolumeAttachmentController) shouldDoDetach(va *longhorn.VolumeAttachm
 	// Currently, the only ticket that is interruptible and frontend disabled is the rebuilding-controller ticket
 	// Offline replica rebuilding has the potential to fail and repeated attempts.
 	// Users can give up (interrupt) the replica rebuilding and attach the degraded volume to a node.
-	if hasInterruptibleAndFrontendDisabledTicket(currentAttachmentTickets) && hasWorkloadTicket(va.Spec.AttachmentTickets) {
+	if hasInterruptibleAndFrontendDisabledTicket(currentAttachmentTickets) && !vol.Status.FrontendDisabled && hasWorkloadTicket(va.Spec.AttachmentTickets) {
```

Comment:
Why do you need this line? Can we directly change `hasUninterruptibleTicket` and remove this line instead?
```
func hasUninterruptibleTicket(attachmentTickets map[string]*longhorn.AttachmentTicket) bool {
	for _, ticket := range attachmentTickets {
		if ticket.Type != longhorn.AttacherTypeSnapshotController &&
			ticket.Type != longhorn.AttacherTypeBackupController &&
			ticket.Type != longhorn.AttacherTypeVolumeRebuildingController {
			return true
		}
	}
	return false
}
```

Reply:
It's better to update the log `log.Info("Workload attachment ticket interrupted snapshot-controller/backup-controller attachment tickets")` as well.

Reply:
`hasUninterruptibleTicket` can handle both `FrontendDisabled=true` and `FrontendDisabled=false` tickets.
`hasInterruptibleAndFrontendDisabledTicket` only focus on `FrontendDisabled=false` tickets.

Yeah, we can merge them together. However, for readability, I prefer to separate them.

Reply:
Merged them together, but separate the checks.